docs(getting-started): add quick-start guide and restructure docs

Gurukiran20 opened the pull request:
This sets up the documentation using MkDocs for the LanceDB + Ray integration project

 What it is Included:
- Created index.md with an overview and guide links
- Added quick-start.md for basic setup and usage
- Added installation.md, usage.md, and contributing.md
- Organized docs under docs/getting-started/
- Added mkdocs.yml configuration for site structure

This setup will help the contributors and users get started quickly and understand how to use and contribute to the project.

chenghao-guo commented on docs/getting-started/quick-start.md:
````diff
@@ -0,0 +1,17 @@
+## Installation
+
+```bash
+pip install lancedb ray
````
maybe pylance should be enough instead of lancedb

jiaoew1991 commented:
> Hi @Gurukiran20 ， Thank you for your contribution!
> 
> I believe the project focuses on lance format + ray, so I believe pylance should be the requirement and lancedb currently not required for this now in the code. cc @jiaoew1991 to take a review as well.

Yes, lancedb is not required, it depends on pylance😁

Gurukiran20 commented:
Resolved merge conflicts and finalized the dev installation section. Ready for review!

jackye1995 commented on docs/usage.md:
this doc is empty?

jackye1995 commented on README.md:
we probably want to move the README to the user guide + contribution guide

jackye1995 commented on docs/index.md:
```diff
@@ -0,0 +1,36 @@
+# Read the dataset back as a Ray Dataset
+ds = read_lance("example.lance")
+
+print(ds.take(3))
```
missing end backquotes

jackye1995 commented on docs/examples.md:
```diff
@@ -0,0 +1,45 @@
+# Examples
+
+## Basic Usage
```
make sure the examples in README are all covered here, this seems to be missing the advanced example, the basic one is also not as detailed. For example, the filter example does not print out "print(f"Filtered count: {filtered_ds.count()}")" as the README does.

jackye1995 commented on docs/api_reference.md:
```diff
@@ -0,0 +1,13 @@
+# API Reference
```
let's remove this page. We should build the API reference to `readthedocs`.

jackye1995 commented on docs/installation.md:
```diff
@@ -0,0 +1,25 @@
+# Installation
+
+## Basic Installation
```
installation should be separated to 

```
pip install lance-ray
```

directly in quick start, and the remaining in contributing guide.

Gurukiran20 commented:
A special thanks to @jackye1995  your guidance, patience, and kindness throughout this contribution meant a lot to me. 🙏
You genuinely treated me with a mentor’s care (almost like a parent teaching a child!), and I learned so much from your feedback and support.
Thank you for making my major open-source documentation task such a great experience! 😊

jackye1995 commented on docs/contributing.md:
```diff
@@ -0,0 +1,44 @@
+
+We welcome contributions from the community!
+
+## How to Contribute
```
we are doing an effort to currently move all the sub-project docs to show up in the main Lance website https://lancedb.github.io/lance. What that means is that you can run the website independently here, but when we publish the website, we can copy the contents here to the Lance website in CI to build the final website. So I think it is probably better if we align the contents in the expected way.

What that means is for contributing page, we just need what is specific for this project. You can imagine that this page will show up under https://lancedb.github.io/lance/community/lance-ray, so it assumes people already have read https://lancedb.github.io/lance/community/, and there is no need to repeat whatever is already there.

jackye1995 commented on docs/contributing.md:
````diff
@@ -0,0 +1,44 @@
+To run all tests using [pytest](https://docs.pytest.org/):
+
+```bash
+pytest
````
should use `uv run pytest`

jackye1995 commented on docs/contributing.md:
````diff
@@ -0,0 +1,44 @@
+```bash
+git clone https://github.com/<your-username>/lance-ray.git
+cd lance-ray
+pip install -e .[dev]
````
uv pip install -e .[dev]

justinrmiller commented on docs/contributing.md:
```diff
@@ -0,0 +1,26 @@
+
+- Python >= 3.8
+
+- Ray >= 2.0.0
```
I believe the project requires Ray 2.40.0 or greater? May want to tighten this and other dependencies?

Reference from docs:

```Python >= 3.10
Ray >= 2.40.0
PyLance >= 0.30.0
lance-namespace >=0.0.5
PyArrow >= 17.0.0
Pandas >= 2.2.0
NumPy >= 2.0.0```

justinrmiller commented on mkdocs.yml:
```diff
@@ -0,0 +1,9 @@
+site_name: Lance-Ray Documentation
```
Looks like this is the referenced documentation? https://github.com/lancedb/lance-spark/blob/main/docs/mkdocs.yml

@Gurukiran20 Let me know if you'd like some help on this, would be great to pair on it.

justinrmiller reviewed (commented):
Looks like a great start, added a couple of comments.

Gurukiran20 commented:
Hi @jackye1995 and @justinrmiller   thanks for pointing that out and offering to help! I’ve restored the .gitignore and tightened up the dependency section in contributing.md. Everything is now aligned with the existing structure.

chenghao-guo commented:
> Hi sir @jackye1995 and @chenghao-guo Just checking in to see if there's anything else you'd like me to change on this PR

Hi @Gurukiran20 , thank you for your contribution.  I have just been back to office recently and reviewed the context. There is something to be done regarding the mkdocs.yaml.

I have cloned your code and set up with mkdocs, then compare it with the lance-spark project as @jackye1995  mentioned. I think there are probably some style difference, so you may refer lance-spark's mkdocs.yaml as @justinrmiller mentioneds on the color style and logo. For example

**Current mkdocs.yaml**
The yellow is not so " lance blue" like other project
<img width="2936" height="1754" alt="image" src="https://github.com/user-attachments/assets/69cd93d9-6a5e-4375-8f8d-68d84e1a3c53" />
<img width="2574" height="1604" alt="image" src="https://github.com/user-attachments/assets/ae418492-bc2e-4d31-b146-4b54353a914a" />

The quick start has a 404.
<img width="1942" height="460" alt="image" src="https://github.com/user-attachments/assets/91af3be8-dd07-4686-964e-2dd5ffaf762a" />



The existing
**lance-spark project**
<img width="2860" height="1720" alt="image" src="https://github.com/user-attachments/assets/28c69320-c87e-462d-a9b9-a0299fbafe0e" />

, consider the lance-ray theme like 
```
theme:
  name: material
  palette:
    - scheme: default
      primary: indigo
      accent: indigo
      toggle:
        icon: material/brightness-7
        name: Switch to dark mode
    - scheme: slate
      primary: indigo
      accent: indigo
      toggle:
        icon: material/brightness-4
        name: Switch to light mode

```

So it would be much better if we can sync the style and fix the some link error, so Jack shall be fine that that.

Gurukiran20 commented:
Hi sir @chenghao-guo and @jackye1995 

Thank you both for your guidance and support throughout this documentation update journey — I truly appreciate the help. 🙏

I’ve now updated the mkdocs.yaml to better match the Lance blue theme and design consistency across projects, as suggested:

- Changed the color palette from light blue to the deeper Lance blue.

- Synced theme settings with the lance-ray style (using indigo for primary and accent in both light and dark modes).

- Fixed the Quick Start 404 link issue.

- Added the GitHub button in the header.

- Included social links (Discord, Twitter, GitHub).

- Updated the website logo and browser tab icon.

Please let me know if there’s anything else you’d like me to add or adjust

Gurukiran20 commented:
Thank you so much, sir🙏 It truly means a lot to be acknowledged as a co-author on the commit. 
As I’m a final year B.Tech student in AI & ML and also interning at Cydra Tech. Over the past two weeks, I had exams, internship Work presentations, and assignments, so I couldn’t contribute actively. From next week onwards, I’ll continue contributing to LanceDB with full focus. 

Since July (past 1.5 months), this journey with LanceDB has been an incredible learning experience for me as a fresher. @chenghao-guo and @jackye1995 your guidance has helped me a lot 🙌 and I sincerely thank you both from the bottom of my heart🙏 .

Gurukiran20 commented:
Hi @jackye1995, @chenghao-guo , and @justinrmiller Good morning

It’s been a while since I last contributed, and I’m excited to be active again in the Lance-Ray project. I’m grateful that my documentation work from #25 was incorporated into #32, especially commits like 2e37d65 (“Add mkdocs integration”) and c181e6e (“use uv for doc”), which reflected the feedback we iterated on together.

However, I noticed that while @jackye1995 mentioned “marked you as coauthor” when closing #25, my name/email (Gurukiran20 gurusm1443@gmail.com
) doesn’t appear in those commits and instead shows up on 4e5d773 (PyPI workflow), which I didn’t work on. Since the PR was closed (not merged), my contribution also isn’t visible under my profile’s contribution history.

Would you be open to amending 2e37d65 or c181e6e with:
Co-authored-by: Gurukiran20 <gurusm1443@gmail.com>  

This would ensure proper attribution in Git history and visibility in my open-source profile.
